### ISSUE TITLE
[Bug]: Cherrypick and Import don't check branch protection

### ISSUE THREAD

**Issue by idanovo** [treeverse/lakeFS#7208]
### What happened?

Cherrypick and Import don't check branch protection

**Comment by N-o-Z** [comment]
Seems like the issue I was talking about (https://github.com/treeverse/lakeFS/issues/7029) is not related.
We might need @treeverse/product's input on this.
For cherry-pick the case seems more clear (do as git does).
Import on the other hand - creates a merge commit, and since there is no corresponding git operation, this might need to be a product decision

**Comment by idanovo** [comment]
I want to clarify the difference between `cherry-pick` in lakeFS and git. As far as I understand, `cherry-pick` in git only stages the changes from a commit to a branch, while in lakeFS, it creates a new commit on our branch. This explains why `cherry-pick` to a protected branch is permitted in git, even though the push command will fail. However, it doesn't make sense in our case.

**Comment by johnnyaug** [comment]
@idanovo, I don't believe this is true: cherry pick in Git does create a commit by default.

**Comment by idanovo** [comment]
> @idanovo, I don't believe this is true: cherry-pick in Git does create a commit by default.

@johnnyaug Sorry, you're right, it creates a new commit by default, but it doesn't push it, while in our case, there's no such thing (AFAIK), Commit == Commit & Push.
